modules/nextflow/src/main/groovy/nextflow/Session.groovy

```diff
@@ -424,6 +432,27 @@ class Session implements ISession {
         return this
     }
 
+    /**
+     * Downloads the custom fsync file from the specified URL.
+     */
+    void downloadCustomFsync() {
+        Path customFsyncPath = workDir.resolve("custom_fsync")
+        if (customFsyncPath.exists()) {
+            log.debug "Custom fsync was already downloaded"
+            return
+        }
+
+        try {
+            URLConnection conn = new URL(CUSTOM_FSYNC_URL).openConnection()
+            conn.connect()
+            Files.copy(conn.inputStream, customFsyncPath)
+            customFsyncPath.setExecutable(true, false)
+            log.info "File downloaded successfully: $customFsyncPath"
+        } catch (IOException e) {
+            log.error "Failed to download file: $CUSTOM_FSYNC_URL", e
+        }
+    }
+
```

Comment:
Should add retries on failure

Reply:
You can just use the HTTPRetryClient that we have for this

Reply:
resolve by copying bin into the image directly